API-Samples/C#/ArmBased/VideoIndexerClient/VideoIndexerClient.cs

```diff
@@ -0,0 +1,331 @@
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Get Information about the Account
+        /// </summary>
+        /// <param name="accountName"></param>
+        /// <returns></returns>
+        public async Task<Account> GetAccount(string accountName)
+        {
+            if (_account != null)
+            {
+                return _account;
+            }
+            Console.WriteLine($"Getting account {accountName}.");
+            try
+            {
+                // Set request uri
+                var requestUri = $"{AzureResourceManager}/subscriptions/{SubscriptionId}/resourcegroups/{ResourceGroup}/providers/Microsoft.VideoIndexer/accounts/{accountName}?api-version={ApiVersion}";
+                var client = new HttpClient(new HttpClientHandler());
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _armAccessToken);
+
+                var result = await client.GetAsync(requestUri);
+
+                result.VerifyStatus(System.Net.HttpStatusCode.OK);
+                var jsonResponseBody = await result.Content.ReadAsStringAsync();
+                var account = JsonSerializer.Deserialize<Account>(jsonResponseBody);
+                VerifyValidAccount(account, accountName);
+                Console.WriteLine($"[Account Details] Id:{account.Properties.Id}, Location: {account.Location}");
+                _account = account;
+                return account;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Uploads a video and starts the video index. Calls the uploadVideo API (https://api-portal.videoindexer.ai/api-details#api=Operations&operation=Upload-Video)
+        /// </summary>
+        /// <param name="videoUrl"> Link To Publicy Accessed Video URL</param>
+        /// <param name="videoName"> The Asset name to be used </param>
+        /// <param name="exludedAIs"> The ExcludeAI list to run </param>
+        /// <param name="waitForIndex"> should this method wait for index operation to complete </param>
+        /// <returns> Video Id of the video being indexed, otherwise throws excpetion</returns>
+        public async Task<string> UploadUrl(string videoUrl , string videoName, string exludedAIs = null, bool waitForIndex = false )
+        {
+            if (_account == null)
+            {
+                throw new Exception("Call Get Account Details First");
+            }
+
+            Console.WriteLine($"Video for account {_account.Properties.Id} is starting to upload.");
+            var content = new MultipartFormDataContent();
+            
+            try
+            {
+                //Build Query Parameter Dictionary
+                var queryDictionary = new Dictionary<string, string>
+                {
+                    { "accessToken", _accountAccessToken },

```

Comment:
Do we need the `accessToken` in the query? It is added as a Header in the constructor of the HttpClient

Reply:
done. old code that was not cleaned

Reply:
@yehiyam  : actually I want it for visibility and the ability to the take the url in debug mode and copy paste it as is to Postman.
it also reminds the users of authentication that must be passed, and not hides it under the utility class.
I have removed the HttpClient(token) constructor